src/awakener.rs

```diff
@@ -0,0 +1,95 @@
+use std::io;
+
+use crate::{poll, sys, Registry, Token};
+
+/// Awakener allows cross-thread waking of [`Poll`].
+///
+/// When created it will cause events with [`Ready::readable()`] and the provided
+/// `token` if [`wake`] is called, possibly from another thread.
+///
+/// # Notes
+///
+/// The `Awakener` needs to be kept alive as long as wake up notifications are
```

Comment:
I would reword this as stating that wake up events are only guaranteed to be delivered while the `Awakener` value is alive.

Reply:
The implementation detail isn't relevant for the public API docs imo

Reply:
Keeping the mentioning of it being an implementation detail (second line)?